src/main/java/com/neuvector/Scanner.java

```diff
@@ -329,6 +341,63 @@ private static String maskCredential(String message, String credential){
         return message.replace(credential, "******");
     }
 
+    static String[] getDockerGroupCmdArgs(String scanReportPath) {
+        String[] cmdGroupArgs = {"", ""};
+        if (!scanResultsFileExist(scanReportPath) || !isRootFile(scanReportPath)) {
+            cmdGroupArgs[0] = "-u";
+            cmdGroupArgs[1] = executeCommand("id -u");
+            return cmdGroupArgs;
+        }
+        return cmdGroupArgs;
+    }
+
+    private static boolean isRootFile(String scanReportPath)  {
+        File file = new File(scanReportPath);
+        try {
+            return scanResultsFileExist(scanReportPath) && getUserPrincipal(scanReportPath, file).getName().equals("root");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String executeCommand(String command) {
+        StringBuilder output = new StringBuilder();
+        try {
+            Process proc = Runtime.getRuntime().exec(command);
+            getExecValueFromBuffer(output, proc);
+        } catch (Exception e) {
```

Comment:
If an exception is thrown here, we just swallow it. In this case, we will be passing just the `-u` parameter to the container. We should avoid this. In other words, if we return `null` from this method, we should not be passing:

```
cmdGroupArgs[0] = "-u";
cmdGroupArgs[1] = executeCommand("id -u");
```

either.

Reply:
7b6ef73de46c3e036fa83da7e263ada1fb205035